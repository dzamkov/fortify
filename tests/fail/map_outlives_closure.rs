use fortify::*;

fn main() {
    let fortified = Fortify::new(456);
    let x = Box::new(123);
    let fortified = fortified.map(|_| Lowered::new(&*x));
    drop(x);
    assert_eq!(**fortified.borrow(), 123);
}