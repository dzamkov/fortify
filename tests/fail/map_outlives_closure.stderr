error[E0505]: cannot move out of `x` because it is borrowed
 --> tests/fail/map_outlives_closure.rs:7:10
  |
6 |     let fortified = fortified.map(|_| Lowered::new(&*x));
  |                                   ---               -- borrow occurs due to use in closure
  |                                   |
  |                                   borrow of `*x` occurs here
7 |     drop(x);
  |          ^ move out of `x` occurs here
8 |     assert_eq!(**fortified.borrow(), 123);
  |                  ------------------ borrow later used here
